flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt

```diff
@@ -66,36 +66,57 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
     }
 
     SentryAndroid.init(context) { options ->
-      // TODO: check if args exist before assigning the values
-
-      options.dsn = args["dsn"] as String?
-      options.isDebug = args["debug"] as Boolean
-      options.environment = args["environment"] as String?
-      options.release = args["release"] as String?
-      options.dist = args["dist"] as String?
-      options.isEnableSessionTracking = args["enableAutoSessionTracking"] as Boolean
-      options.sessionTrackingIntervalMillis = (args["autoSessionTrackingIntervalMillis"] as Int).toLong()
-      options.anrTimeoutIntervalMillis = (args["anrTimeoutIntervalMillis"] as Int).toLong()
-      // expose options for isAttachThreads?
-      options.isAttachStacktrace = args["attachStacktrace"] as Boolean
-
-      val enableAutoNativeBreadcrumbs = args["enableAutoNativeBreadcrumbs"] as Boolean
-      options.isEnableActivityLifecycleBreadcrumbs = enableAutoNativeBreadcrumbs
-      options.isEnableAppLifecycleBreadcrumbs = enableAutoNativeBreadcrumbs
-      options.isEnableSystemEventBreadcrumbs = enableAutoNativeBreadcrumbs
-      options.isEnableAppComponentBreadcrumbs = enableAutoNativeBreadcrumbs
-
-      options.maxBreadcrumbs = args["maxBreadcrumbs"] as Int
-      options.cacheDirSize = args["cacheDirSize"] as Int
-
-      val level = args["diagnosticLevel"] as String
-      val sentryLevel = SentryLevel.valueOf(level.toUpperCase(Locale.ROOT))
-      options.setDiagnosticLevel(sentryLevel)
-
-      val anrEnabled = args["anrEnabled"] as Boolean
+      (args["dsn"] as? String)?.let {
+        options.dsn = it
+      }
+      (args["debug"] as? Boolean)?.let {
+        options.isDebug = it
+      }
```

Comment:
we could move the reading of the `diagnosticLevel` for only when the `debug` flag is enabled, so we save some allocation, good point.